src/components/SearchBar.tsx

```diff
@@ -0,0 +1,60 @@
+import _ from 'lodash'
+import React, { useEffect, useState } from 'react'
+import { StyleSheet, TextInput, View } from 'react-native'
+import { useDispatch, useSelector } from 'react-redux'
+import Search from 'src/icons/Search'
+import { setFilteredVendors, setSearchQuery } from 'src/vendors/actions'
+import { vendorsSelector } from 'src/vendors/selector'
+
+export default function Searchbar() {
+  const dispatch = useDispatch()
+  const vendors = Object.values(useSelector(vendorsSelector))
+  const [search, setSearch] = useState<string>('')
+
+  useEffect(() => {
+    const filteredVendors = vendors.filter(
+      (vendor) =>
+        vendor.title.toLowerCase().includes(search.toLowerCase()) ||
+        _.some(vendor.tags, (tag) => tag.toLowerCase().includes(search.toLowerCase()))
+    )
+    dispatch(setFilteredVendors(filteredVendors))
+  }, [search])
+
+  const handleSearch = (search: string) => {
+    setSearch(search)
+    dispatch(setSearchQuery(search))
+  }
```

Comment:
This logic should be triggered by `vendor/saga.ts`

E.g. `takeEvery(Action.SET_SEARCH_QUERY, watchMapFilter)`
```ts
function* watchMapFilter (action: any) {
  const vendors = filterVendors(action.payload)
  yield put(setFilteredVendors(vendors))
}
```

Just take note of this pattern. With this, you have taken the logic and decoupled it from the component that handles searching.

Reply:
Also please refrain from importing the `_` namespace from `lodash`. Only import the functions you need.

E.g. `import { some } from 'lodash'`